vcd/resource_vcd_edgegateway.go

```diff
@@ -0,0 +1,232 @@
+package vcd
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdEdgeGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdEdgeGatewayCreate,
+		Delete: resourceVcdEdgeGatewayDelete,
+		Read:   resourceVcdEdgeGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"org": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vdc": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"advanced": &schema.Schema{
+				Type:        schema.TypeBool,
+				Required:    true,
+				ForceNew:    true,
+				Description: "True if the gateway uses advanced networking. (Set by default in vCD 9.7+)",
+			},
+			"gateway_configuration": &schema.Schema{
```

Comment:
Looking now again, I am kind'a think even `configuration` even enough. @lvirbalas @Didainius

Reply:
Agree

Reply:
+1 for `configuration`

Reply:
+1

Reply:
Changed